include/bcs/MFEMVectorCurlDirichletBC.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include "MFEMVectorDirichletBCBase.h"
+
+class MFEMVectorCurlDirichletBC : public MFEMVectorDirichletBCBase
```

Comment:
I know this naming convention (`VectorCurlDirichletBC` and `VectorDivDirichletBC`) is consistent with MOOSE's penalty method equivalents, but on reflection I think it's preferable to refer to them as `VectorTangentialDirichletBC` and `VectorNormalDirichletBC` to make it clearer what they actually do